diff --git a/.gitignore b/.gitignore
index 35af0b2..f7002ef 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,4 +1,3 @@
 dist/
 .idea/
-node_modules/
-*.iml
\ No newline at end of file
+node_modules/
\ No newline at end of file
diff --git a/assets/arrows.png b/assets/arrows.png
deleted file mode 100644
index 06d9e3b..0000000
Binary files a/assets/arrows.png and /dev/null differ
diff --git a/assets/bg.png b/assets/bg.png
deleted file mode 100644
index 2b558a0..0000000
Binary files a/assets/bg.png and /dev/null differ
diff --git a/assets/busd.png b/assets/busd.png
deleted file mode 100644
index 3634799..0000000
Binary files a/assets/busd.png and /dev/null differ
diff --git a/assets/checked.png b/assets/checked.png
deleted file mode 100644
index f355e4e..0000000
Binary files a/assets/checked.png and /dev/null differ
diff --git a/assets/dai.png b/assets/dai.png
deleted file mode 100644
index fcf9195..0000000
Binary files a/assets/dai.png and /dev/null differ
diff --git a/assets/logo.png b/assets/logo.png
deleted file mode 100644
index c7bfb77..0000000
Binary files a/assets/logo.png and /dev/null differ
diff --git a/assets/usdt.png b/assets/usdt.png
deleted file mode 100644
index 6f68256..0000000
Binary files a/assets/usdt.png and /dev/null differ
diff --git a/common.js b/common.js
index 7a31f68..04ead7f 100644
--- a/common.js
+++ b/common.js
@@ -8,7 +8,6 @@ function convertBN(val) {
     return cBN(val.toString())
 }
 
-
 async function init_contracts() {
     if (EXT == null){
         console.error("call init_contracts before init extension")
@@ -214,12 +213,16 @@ async function calc_slippage(deposit) {
     var real_values = [...$("[id^=currency_]")].map((x,i) => +($(x).val()));
     var Sr = real_values.reduce((a,b) => a+b, 0);
 
-    var values = real_values.map((x,i) => cBN(Math.floor(x).toString()).toFixed(0,1));
+    let values = new Array(CONFIG.numCoins)
+    for (let i = 0; i < CONFIG.numCoins; i++) {
+        values[i] = valToBN(real_values[i], CONFIG.coinPrecision[i])
+        console.log("values[i]", values[i].toString())
+    }
     var token_amount = await SWAP.methods.calc_token_amount(values, deposit).call(CALL_OPTION);
     var virtual_price = await SWAP.methods.get_virtual_price().call(CALL_OPTION);
 
-    console.log("virtual_price", virtual_price.toString())
-    console.log("token_amount", token_amount.toString())
+    console.log("virtual_price", virtual_price, virtual_price.toString())
+    console.log("token_amount", token_amount, token_amount.toString())
     var Sv = virtual_price.mul(token_amount).div(BN(10).pow(BN(36)));
     console.log("bn Sv", Sv.toString())
     Sv = convertBN(Sv)
diff --git a/config.js b/config.js
index b97c893..137e2f6 100644
--- a/config.js
+++ b/config.js
@@ -1,10 +1,25 @@
-const CONFIG = {
+const MAINNET_CONFIG = {
+    endpoint: "https://api.s0.t.hmny.io",
+    shardID: 1,
+    chainID: 1,
+
+    poolToken: "0x3f8d8f755bbb73e0e8f85a4009501fc5f213fc1e",
+    swapContract: "0xc08e2d250bd031409c9d87888356434c5384f188",
+
+    numCoins: 3,
+    coinPrecision: [1e18, 1e18, 1e18],
+
+    gasPrice: 0x4a817c800,
+    gasLimit: 0x6691b7,
+}
+
+const TESTNET_CONFIG = {
     endpoint: "https://api.s0.b.hmny.io",
     shardID: 0,
     chainID: 2,
 
-    poolToken: "0x3a647d615d6f3fe70826ad68303d953d8633daac",
-    swapContract: "0xb0f725c8c7534dba32ae2870145aa3b2c2fde9dc",
+    poolToken: "0xed2d6c263a0207a5b7512c20a464745d800ccf23",
+    swapContract: "0x486ffe573dea3c47fe9f14be4bd2542616f872e1",
 
     numCoins: 3,
     coinPrecision: [1e18, 1e18, 1e18],
@@ -13,6 +28,8 @@ const CONFIG = {
     gasLimit: 0x6691b7,
 }
 
+const CONFIG = TESTNET_CONFIG
+
 let CALL_OPTION = {
     gasPrice: CONFIG.gasPrice,
     gasLimit: CONFIG.gasLimit,
diff --git a/deposit.html b/deposit.html
index 3e70abe..e75dde6 100644
--- a/deposit.html
+++ b/deposit.html
@@ -21,13 +21,6 @@
     <link rel="icon" type="image/png" sizes="16x16" href="assets/favicon-16x16.png?rev=f30faa950b2b49eeffcaad8d774f6b10">
     <link rel="mask-icon" href="assets/safari-pinned-tab.svg" color="#5bbad5">
     <link rel="stylesheet" href="tvision.css?rev=3eda33f4fa54fe61a0ff034c823d8320" />
-    <script src="https://ajax.googleapis.com/ajax/libs/webfont/1.6.26/webfont.js" type="text/javascript"></script>
-    <script type="text/javascript">
-        WebFont.load({google: {families:
-                    ["Open Sans:300,300italic,400,400italic,600,600italic,700,700italic,800,800italic", "Nunito Sans:regular,700,900", "Nunito:regular,700,800,900"]
-            }});
-    </script>
-
     <script src="https://browser.sentry-cdn.com/5.13.0/bundle.min.js" integrity="sha384-ePH2Cp6F+/PJbfhDWeQuXujAbpil3zowccx6grtsxOals4qYqJzCjeIa7W2UqunJ" crossorigin="anonymous">
     </script>
     <script src='https://browser.sentry-cdn.com/5.13.0/captureconsole.min.js'></script>
@@ -43,8 +36,8 @@
 </head>
 <body>
 <div class="top-menu-bar">
-   <!-- <label for="hamburger" class='border-menu'></label>
-    <input type="checkbox" id="hamburger"/>-->
+    <label for="hamburger" class='border-menu'></label>
+    <input type="checkbox" id="hamburger"/>
 
     <a href="/">Buy and sell</a>
     <a href="/deposit.html">Deposit</a>
@@ -52,31 +45,29 @@
 
 </div>
 <div id="screen">
-    <div class="logo">
-        <h1><img src="assets/logo.png" alt="Harmony One" height="50"></h1>
+    <div class="blue window">
+        <h1><img src="assets/logo.svg" alt="🌀 Curve" height="50"></h1>
     </div>
     <div class="error window half-width info" id="error-window"></div>
 
-    <div class="window add-liquidity">
+    <div class="window white add-liquidity">
         <fieldset class="currencies">
             <legend>Currencies:</legend>
             <ul>
                 <li>
-                    <label for="currency_0">USDT</label>
+                    <label for="currency_0">Token A</label>
                     <input type="text" id="currency_0" disabled name="from_cur" value="0.00">
                 </li>
                 <li>
-                    <label for="currency_1">DAI</label>
+                    <label for="currency_1">Token B</label>
                     <input type="text" id="currency_1" disabled name="from_cur" value="0.00">
                 </li>
                 <li>
-                    <label for="currency_2">BUSD</label>
+                    <label for="currency_2">Token C</label>
                     <input type="text" id="currency_2" disabled name="from_cur" value="0.00">
                 </li>
             </ul>
         </fieldset>
-
-        <fieldset>
         <ul>
             <li>
                 <input id="max-balances" type="checkbox" name="max-balances" checked>
@@ -87,11 +78,10 @@
                 <label for="inf-approval">Infinite approval - trust this contract forever</label>
             </li>
         </ul>
-        </fieldset>
 
         <p style="text-align: center">
             <button id="add-liquidity">Deposit</button>
-        <p class='simple-error' id='highslippage-warning' style="border-radius:8px"><span class='text'>Warning! High slippage</span>: <span class='percent'></span>%</p>
+        <p class='simple-error' id='highslippage-warning'><span class='text'>Warning! High slippage</span>: <span class='percent'></span>%</p>
         <p class='success' id='bonus-window'>Bonus: <span></span>%</p>
         </p>
     </div>
@@ -101,10 +91,10 @@
     <fieldset id="lp-info-currency">
         <legend>Currency reserves</legend>
         <ul id='balances-info'>
-            <li><b>USDT:</b> <span></span></li>
-            <li><b>DAI:</b> <span></span></li>
-            <li><b>BUSD:</b> <span></span></li>
-            <li><b>USDT + DAI + BUSD:</b> <span></span></li>
+            <li><b>Token A:</b> <span></span></li>
+            <li><b>Token B:</b> <span></span></li>
+            <li><b>Token C:</b> <span></span></li>
+            <li><b>A + B + C:</b> <span></span></li>
         </ul>
         <p><b>Fee:</b> <span id='fee-info'></span>%</p>
         <p><b>Admin fee:</b> <span id='admin-fee-info'></span>%</p>
@@ -112,10 +102,10 @@
     <fieldset id="lp-info-container">
         <legend>My share:</legend>
         <ul id='lp-info'>
-            <li><b>USDT:</b> <span></span></li>
-            <li><b>DAI:</b> <span></span></li>
-            <li><b>BUSD:</b> <span></span></li>
-            <li><b>USDT + DAI + BUSD:</b> <span></span></li>
+            <li><b>Token A:</b> <span></span></li>
+            <li><b>Token B:</b> <span></span></li>
+            <li><b>Token C:</b> <span></span></li>
+            <li><b>A + B + C:</b> <span></span></li>
         </ul>
     </fieldset>
 </div>
diff --git a/deposit.js b/deposit.js
index c9ec047..e90c159 100644
--- a/deposit.js
+++ b/deposit.js
@@ -75,7 +75,7 @@ async function init_ui() {
             if (el.val() > BALANCES[i])
                 el.css('background-color', 'red')
             else
-                el.css('background-color', '#036574');
+                el.css('background-color', 'blue');
         }));
     }
 
diff --git a/exchange.js b/exchange.js
index 8076bf7..b38e8f1 100644
--- a/exchange.js
+++ b/exchange.js
@@ -36,7 +36,7 @@ async function highlight_input() {
     if (el.val() > balance)
         el.css('background-color', 'red')
     else
-        el.css('background-color', '#036574');
+        el.css('background-color', 'blue');
 }
 
 function promiseTimeout(ms) {
@@ -51,7 +51,7 @@ async function set_to_amount() {
             $('#to_currency').val(dy);
             let exchange_rate = (dy_ / dx_).toFixed(4);
             if(exchange_rate <= 0.98) $("#to_currency").css('background-color', 'red')
-            else $("#to_currency").css('background-color', '#036574')
+            else $("#to_currency").css('background-color', '#505070')
             if(isNaN(exchange_rate)) exchange_rate = "Not available"
             var default_account = ETH_ADDR;
             let rawBalance = await UL_COINS[toCurrency].methods.balanceOf(default_account).call(CALL_OPTION);
diff --git a/index.html b/index.html
index 0cd3cb2..a00a8f6 100644
--- a/index.html
+++ b/index.html
@@ -1,40 +1,27 @@
 <!doctype html>
 <html>
 <head>
-    <meta charset="UTF-8"/>
-    <meta name="viewport" content="width=device-width, initial-scale=1"/>
-    <meta property="og:title" content="compound.curve.fi"/>
-    <meta property="og:url" content="https://compound.curve.fi"/>
-    <meta property="og:type" content="website"/>
-    <meta property="og:description" content="A curve.fi portal for swapping cDAI/cUSDC"/>
-    <meta property="og:image" content="./assets/compoundcurve_preview.png"/>
-    <meta name="twitter:card" content="summary_large_image"/>
-    <meta name="twitter:title" content="compound.curve.fi"/>
-    <meta name="twitter:site" content="@CurveFinance"/>
-    <meta name="twitter:creator" content="@CurveFinance"/>
-    <meta name="twitter:description" content="A curve.fi portal for swapping cDAI/cUSDC"/>
-    <meta name="twitter:url" content="https://compound.curve.fi"/>
-    <meta name="twitter:image" content="./assets/compoundcurve_preview.png"/>
-    <meta name="viewport" content="width=700" />
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta property="og:title" content="compound.curve.fi" />
+    <meta property="og:url" content="https://compound.curve.fi" />
+    <meta property="og:type" content="website" />
+    <meta property="og:description" content="A curve.fi portal for swapping cDAI/cUSDC" />
+    <meta property="og:image" content="./assets/compoundcurve_preview.png" />
+    <meta name="twitter:card" content="summary_large_image" />
+    <meta name="twitter:title" content="compound.curve.fi" />
+    <meta name="twitter:site" content="@CurveFinance" />
+    <meta name="twitter:creator" content="@CurveFinance" />
+    <meta name="twitter:description" content="A curve.fi portal for swapping cDAI/cUSDC" />
+    <meta name="twitter:url" content="https://compound.curve.fi" />
+    <meta name="twitter:image" content="./assets/compoundcurve_preview.png" />
     <title>Curve.fi :: Compounded</title>
     <link rel="apple-touch-icon" sizes="180x180" href="assets/apple-touch-icon.png?">
     <link rel="icon" type="image/png" sizes="32x32" href="assets/favicon-32x32.png?">
     <link rel="icon" type="image/png" sizes="16x16" href="assets/favicon-16x16.png?">
     <link rel="mask-icon" href="safari-pinned-tab.svg" color="#5bbad5">
-    <link rel="stylesheet" href="tvision.css?rev=3eda33f4fa54fe61a0ff034c823d8320"/>
-    <script src="https://ajax.googleapis.com/ajax/libs/webfont/1.6.26/webfont.js" type="text/javascript"></script>
-    <script type="text/javascript">
-        WebFont.load({
-            google: {
-                families:
-                    ["Open Sans:300,300italic,400,400italic,600,600italic,700,700italic,800,800italic", "Nunito Sans:regular,700,900", "Nunito:regular,700,800,900"]
-            }
-        });
-    </script>
-
-
-    <script src="https://browser.sentry-cdn.com/5.13.0/bundle.min.js"
-            integrity="sha384-ePH2Cp6F+/PJbfhDWeQuXujAbpil3zowccx6grtsxOals4qYqJzCjeIa7W2UqunJ" crossorigin="anonymous">
+    <link rel="stylesheet" href="tvision.css?rev=3eda33f4fa54fe61a0ff034c823d8320" />
+    <script src="https://browser.sentry-cdn.com/5.13.0/bundle.min.js" integrity="sha384-ePH2Cp6F+/PJbfhDWeQuXujAbpil3zowccx6grtsxOals4qYqJzCjeIa7W2UqunJ" crossorigin="anonymous">
     </script>
     <script src='https://browser.sentry-cdn.com/5.13.0/captureconsole.min.js'></script>
     <script src="jquery.min.js"></script>
@@ -49,134 +36,86 @@
 </head>
 <body>
 <div class="top-menu-bar">
-    <!-- <label for="hamburger" class='border-menu'></label>
-     <input type="checkbox" id="hamburger"/>-->
+    <label for="hamburger" class='border-menu'></label>
+    <input type="checkbox" id="hamburger"/>
 
     <a href="/">Buy and sell</a>
     <a href="/deposit.html">Deposit</a>
     <a href="/withdraw.html">Withdraw</a>
 </div>
 <div id="screen">
-    <div class="logo">
-        <h1><img src="assets/logo.png" alt="Harmony One" height="50"></h1>
+    <div class="blue window">
+        <h1><img src="assets/logo.svg" alt="🌀 Curve" height="50"></h1>
     </div>
     <div class="error window half-width info" id="error-window"></div>
 
-    <div class="window">
+    <div class="window white">
         <div style="display: table; margin: auto">
-            <fieldset class="exchange-panel">
+            <fieldset style="float: left">
                 <legend>From:</legend>
-                <div class='maxbalance'>Max: <span></span></div>
+                <div class='maxbalance'>Max: <span></span> </div>
                 <ul>
                     <li>
                         <input type="text" id="from_currency" disabled name="from_currency" value='0.00'>
                     </li>
-                    <li class="li-flex">
+                    <li>
                         <input type="radio" id="from_cur_0" name="from_cur" value="0">
-                        <label for="from_cur_0">
-                            <div class="usdt-icon"></div>
-                            <div class="label-text">
-                                USDT
-                            </div>
-                        </label>
+                        <label for="from_cur_0">Token A</label>
                     </li>
-                    <li class="li-flex">
+                    <li>
                         <input type="radio" id="from_cur_1" name="from_cur" value="1">
-                        <label for="from_cur_1">
-                            <div class="dai-icon"></div>
-                            <div class="label-text">
-                                DAI
-                            </div>
-                        </label>
+                        <label for="from_cur_1">Token B</label>
                     </li>
-                    <li class="li-flex">
-
+                    <li>
                         <input type="radio" id="from_cur_2" name="from_cur" value="2">
-                        <label for="from_cur_2">
-                            <div class="busd-icon"></div>
-                            <div class="label-text">
-                                BUSD
-                            </div>
-                        </label>
+                        <label for="from_cur_2">Token C</label>
                     </li>
                 </ul>
             </fieldset>
-
-            <fieldset class="exchange-panel">
-                <div class="arrows">
-                    <img src="assets/arrows.png">
-                </div>
-            </fieldset>
-
-            <fieldset class="exchange-panel">
+            <fieldset style="float: left">
                 <legend>To:</legend>
-                <div class='maxbalance'>Max: <span></span></div>
+                <div class='maxbalance'>Max: <span></span> </div>
                 <ul>
                     <li>
                         <input type="text" id="to_currency" name="to_currency" value="0.00" disabled>
                     </li>
-                    <li class="li-flex">
+                    <li>
                         <input type="radio" id="to_cur_0" name="to_cur" value="0">
-                        <label for="to_cur_0">
-                            <div class="usdt-icon"></div>
-                            <div class="label-text">
-                                USDT
-                            </div>
-                        </label>
+                        <label for="to_cur_0">Token A</label>
                     </li>
-                    <li class="li-flex">
+                    <li>
                         <input type="radio" id="to_cur_1" name="to_cur" value="1">
-                        <label for="to_cur_1">
-                            <div class="dai-icon"></div>
-                            <div class="label-text">
-                                DAI
-                            </div>
-                        </label>
+                        <label for="to_cur_1">Token B</label>
                     </li>
-                    <li class="li-flex">
+                    <li>
                         <input type="radio" id="to_cur_2" name="to_cur" value="2">
-                        <label for="to_cur_2">
-                            <div class="busd-icon"></div>
-                            <div class="label-text">
-                                BUSD
-                            </div>
-                        </label>
+                        <label for="to_cur_2">Token C</label>
                     </li>
                 </ul>
             </fieldset>
             <div class='clearfix'></div>
-
             <p class='exchange-rate'>Exchange rate (including fees): <span id="exchange-rate"></span></p>
-
-            <fieldset>
-                <br/>
-                <div id='max_slippage'><span>Max slippage:</span>
-
-
-                    <input id="slippage05" type="radio" name="slippage" value='0.005'>
-                    <label for="slippage05">0.5%</label>
-
-                    <input id="slippage1" type="radio" name="slippage" checked value='0.01'>
-                    <label for="slippage1">1%</label>
-
-                    <input id="custom_slippage" type="radio" name="slippage" value='-'>
-
-                    <label for="custom_slippage" style="margin-left:4px;height:14px;position:absolute;">
-                        <input type="text" id="custom_slippage_input" disabled name="custom_slippage_input"> %
-                    </label>
-
-                </div>
-                <ul>
-                    <li>
-                        <input id="inf-approval" type="checkbox" name="inf-approval" checked>
-                        <label for="inf-approval">Infinite approval - trust this contract forever</label>
-                    </li>
-                </ul>
-                <p class='trade-buttons'>
-                    <button id="trade">Sell</button>
-                </p>
-
-            </fieldset>
+            <div id='max_slippage'><span>Max slippage:</span>
+                <input id="slippage05" type="radio" name="slippage" value='0.005'>
+                <label for="slippage05">0.5%</label>
+
+                <input id="slippage1" type="radio" name="slippage" checked value='0.01'>
+                <label for="slippage1">1%</label>
+
+                <input id="custom_slippage" type="radio" name="slippage" value='-'>
+                <label for="custom_slippage">
+                    <input type="text" id="custom_slippage_input" disabled name="custom_slippage_input"> %
+                </label>
+            </div>
+            <ul>
+                <li>
+                    <input id="inf-approval" type="checkbox" name="inf-approval" checked>
+                    <label for="inf-approval">Infinite approval - trust this contract forever</label>
+                </li>
+            </ul>
+            <p class='trade-buttons'>
+                <button id="trade">Sell</button>
+            </p>
         </div>
     </div>
 </div>
@@ -185,10 +124,10 @@
     <fieldset id="lp-info-currency">
         <legend>Currency reserves</legend>
         <ul id='balances-info'>
-            <li><b>USDT:</b> <span></span></li>
-            <li><b>DAI:</b> <span></span></li>
-            <li><b>BUSD:</b> <span></span></li>
-            <li><b>USDT + DAI + BUSD:</b> <span></span></li>
+            <li><b>Token A:</b> <span></span></li>
+            <li><b>Token B:</b> <span></span></li>
+            <li><b>Token C:</b> <span></span></li>
+            <li><b>A + B + C:</b> <span></span></li>
         </ul>
         <p><b>Fee:</b> <span id='fee-info'></span>%</p>
         <p><b>Admin fee:</b> <span id='admin-fee-info'></span>%</p>
@@ -196,10 +135,10 @@
     <fieldset id="lp-info-container">
         <legend>My share:</legend>
         <ul id='lp-info'>
-            <li><b>USDT:</b> <span></span></li>
-            <li><b>DAI:</b> <span></span></li>
-            <li><b>BUSD:</b> <span></span></li>
-            <li><b>USDT + DAI + BUSD:</b> <span></span></li>
+            <li><b>Token A:</b> <span></span></li>
+            <li><b>Token B:</b> <span></span></li>
+            <li><b>Token C:</b> <span></span></li>
+            <li><b>A + B + C:</b> <span></span></li>
         </ul>
     </fieldset>
 </div>
diff --git a/tvision.css b/tvision.css
index d96a262..376b52c 100644
--- a/tvision.css
+++ b/tvision.css
@@ -1,247 +1,231 @@
-/* nunito-regular - latin */
-*:focus {
-    outline: none !important;
-    box-shadow: none;
-}
-
-.p-c {
-    color: #036574
-}
-
 html {
-    background-image: url(assets/bg.png);
-    background-position: top;
-    background-repeat: round;
-    background-attachment: fixed;
-    width: 100%;
-    font-family: Nunito, sans-serif !important;
-    font-weight: bold;
-    font-size: 10pt
+  background: #a5a4ce;
+  font-family: System, monospace;
+  font-size: 10pt
 }
 
 body {
     padding: 0;
     margin: 0;
-    text-align: center;
-}
-
-.dai-icon {
-    height:20px;
-    display: inline-block;
-    width: 24px;
-    background: red;
-    padding: 0;
-    margin: 0;
-    background: url(assets/dai.png);
-    background-size: 20px;
-    background-repeat: no-repeat;
-    background-position: left;
-}
-
-.usdt-icon {
-    height:20px;
-    display: inline-block;
-    width: 24px;
-    background: red;
-    padding: 0;
-    margin: 0;
-    background: url(assets/usdt.png);
-    background-size: 20px;
-    background-repeat: no-repeat;
-    background-position: left;
-}
-
-.busd-icon {
-    display: inline-block;
-    width: 24px;
-    height:20px;
-    background: red;
-    padding: 0;
-    margin: 0;
-    background: url(assets/busd.png);
-    background-size: 20px;
-    background-repeat: no-repeat;
-    background-position: left;
-}
-
-.label-text {
-    display: inline-block;
-    vertical-align: top;
-    height:20px;
-}
-
-.arrows {
-    padding:10px;
-}
-
-.arrows img {
-    width:40px;
-    margin-top:80px;
-}
-
-.logo {
-    text-align: center;
-    margin-top: 90px;
 }
 
 .window {
-    position: relative;
-    cursor: default;
-    margin: 2em auto;
-    width: 80%;
-    background: rgba(255, 255, 255, .5);
-    opacity: 1;
-    color: black;
-    border: none;
-    padding: 1em;
-    border-radius: 8px;
-    max-width: 640px;
+  position: relative;
+  cursor: default;
+  margin: 2em auto; width: 80%;
+  background: #3465a4; color: white;
+  border: 6px double white;
+  padding: 1em;
+  box-shadow:
+    0   0   0   3px #3465a4,
+    1em 1em 3px 0 rgba(0,0,0,.5);
+  max-width: 640px;
 }
 
-
 .window.error {
     background: red;
-    box-shadow: 0 0 0 3px red,
-    1em 1em 3px 0 rgba(0, 0, 0, .5);
+    box-shadow:
+        0   0   0   3px red,
+        1em 1em 3px 0 rgba(0,0,0,.5);
     display: none;
 }
 
 .window.white {
-    background: silver;
-    color: black;
-    border-color: white;
-    box-shadow: 0 0 0 3px silver,
-    1em 1em 3px 0 rgba(0, 0, 0, .5);
+  background: silver;
+  color: black;
+  border-color: white;
+  box-shadow:
+    0   0   0   3px silver,
+    1em 1em 3px 0 rgba(0,0,0,.5);
 }
 
 .window.half-width {
-    width: 80%;
+  width: 80%;
 }
 
 .window :link {
-    color: inherit;
-    font-weight: bolder;
-    text-decoration: none;
+  color: inherit;
+  font-weight: bolder;
+  text-decoration: none;
 }
 
 .window :link:hover {
-    text-decoration: underline;
+  text-decoration: underline;
 }
 
 .window h1 {
-    text-align: center;
+  text-align: center;
 }
 
 
 .window.blue p {
-    /*color: white; text-shadow: 0 0 2px white;*/
+  /*color: white; text-shadow: 0 0 2px white;*/
 }
 
 button {
-    margin: 0;
-    padding: 0;
-    border: none;
-    cursor: pointer;
+  margin:0;padding:0;border:none;
+  cursor: pointer;
 }
 
 button,
 .button,
 .button:link {
-    display: inline-block;
-    font-weight: normal;
-    color: white;
-    font-family: inherit;
-    font-size: 16px;
-    background: transparent;
-    padding: 0 1em;
-    position: relative;
-    border: 1px solid white;
-    border-radius: 16px;
+  display: inline-block;
+  font-weight: bolder;
+  color: white;
+  font-family: inherit;
+  font-size: inherit;
+  background: green;
+  padding: 0 1em;
+  box-shadow: .5em .5em 3px rgba(0,0,0,.5);
+  position: relative;
+  transition: .2s all ease-in;
+  border: 1px outset green;
+}
+
+button.full-width,
+.button.full-width {
+  display: block;
+  text-align: center;
+}
+
+button:before,
+.button:before {
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  content: "";
+  position: absolute;
+  /*background: rgba(255,255,0,.5); /* DEBUG CLICK AREA */
+}
+
+button:hover,
+.button:active,
+.button:focus,
+.button:hover
+{
+  transform: translate3d(.25em,.25em,10em);
+  box-shadow: 0 0 0 rgba(0,0,0,.5);
+  background: limegreen;
+}
+
+button:hover:before,
+.button:hover:before,
+button:focus:before,
+.button:focus:before
+{
+  top: -1em;
+  left: -1em;
+}
+
+.button.pressed {
+    box-shadow: none;
+    left: 0.25em;
+    top: 0.25em;
+    margin-right: 0.25em;
+    background: gray;
+    border: 1px outset #707070;
 }
 
+button:first-letter,
+.button:first-letter {
+  color: yellow;
+}
 
 input, textarea {
-    background: #036574 !important;
-    color: white;
-    width: 100%;
-    max-width: 100%;
-    font-size: 12pt;
-    border: none !important;
-    padding: 3px !important;
+  background: blue;
+  color: aqua;
+  width: 100%;
+  max-width: 100%;
+  font-size: 12pt;
 }
 
 input {
-    padding: 3px;
-    text-align: center;
+  padding: 1px 0 1px 0;
+  border-radius: 0;
 }
 
 input:disabled {
-    background: #036574 !important;
-    color: white;
-    border: none;
+  background: #505070;
+  color: #d0d0d0;
+  border: 2px solid #505070;
 }
 
 input[type=radio],
 input[type=checkbox] {
-    display: inline-block;
-    background-color: white;
-    width:12px;
-    height:12px;
+  display: none;
+}
+
+input[type=radio] + label[for]:before,
+input[type=checkbox] + label[for]:before {
+  text-align:center;
+  display: inline-block;
+  height: 1em; width: 2em;
+}
+
+input[type=radio] + label[for]:before {
+  content: "( ) ";
+  cursor: pointer;
+}
+
+input[type=checkbox] + label[for]:before {
+  content: "[ ] ";
+  cursor: pointer;
+}
+
+input[type=radio]:checked + label[for]:before {
+  content: "(•) ";
+  cursor: pointer;
+}
+
+input[type=checkbox]:checked + label[for]:before {
+  content: "[X] ";
+  cursor: pointer;
 }
 
 .window > textarea {
-    width: 100%;
-    border: none;
+  width: 100%;
+  border: none;
 }
 
 
 ul {
-    margin: 0;
-    padding: 0;
+  margin: 0;
+  padding: 0;
 }
 
 li {
-    margin: 0;
-    padding: 0;
-    list-style: none;
-    line-height: 30px;
-    text-align: left;
-}
-
-.li-flex {
-    display: flex;
-    padding: 8px 8px 8px 0;
+  margin: 0;
+  padding: 0;
+  list-style: none;
 }
 
 .top-menu-bar {
-    top: 0;
-    left: 0;
-    right: 0;
-    background: #036574;
-    color: white;
-    font-weight: bolder;
-    text-align: center;
+  top: 0; left: 0; right: 0;
+  background: silver;
+  font-weight: bolder;
+  text-align: center;
 }
 
 .top-menu-bar a, .poolsdialog a .top-menu-bar .icon-container {
-    margin: 0 1em;
-    text-decoration: none;
-    color: white;
-    padding: 0 1em;
+  margin: 0 1em;
+  text-decoration: none;
+  color: black;
+  padding: 0 1em;
 }
 
 .top-menu-bar a:visited, .poolsdialog a:visited {
-    color: white;
+    color: black;
 }
 
 .top-menu-bar a:hover, .poolsdialog div.selected, .poolsdialog div.selected a, .poolsdialog div.selected a:visited {
-    background-color: rgba(255,255,255,.5);
-    color: white;
+    background-color: #505050;
+    color: #c0c0c0;
 }
 
-.top-menu-bar a.selected, .poolsdialog div:hover,
-.poolsdialog div:hover a, .poolsdialog div:hover a:visited {
-    background-color: rgba(255,255,255,.5);
+.top-menu-bar a.selected, .poolsdialog div:hover, .poolsdialog div:hover a, .poolsdialog div:hover a:visited {
+    background-color: blue;
     color: white;
 }
 
@@ -249,39 +233,20 @@ li {
     display: block;
 }
 
-.exchange-panel {
-    float:left
-}
-
-/*
-@media only screen and (min-device-width: 320px) and (max-device-width: 480px) {
-    .exchange-panel {
-        float:none;
-    }
-
-    .arrows {
-        display: none;
-    }
-
-   !* .top-menu-bar a, .top-menu-bar .icon-container {
-        display: block;
-        margin: 0;
-        padding: 0.5em 0;
-    }*!
+@media only screen and (min-device-width : 320px) and (max-device-width : 480px) {
+  .top-menu-bar a, .top-menu-bar .icon-container {
+    display: block;
+    margin: 0;
+    padding: 0.5em 0;
+  }
 }
-*/
 
 div.add-liquidity input {
     margin-bottom: 0.5em;
 }
 
-strong:before, strong:after {
-    content: "*";
-}
-
-i:before, i:after {
-    content: "/";
-}
+strong:before, strong:after { content: "*"; }
+i:before, i:after { content: "/"; }
 
 .trade-buttons {
     text-align: center;
@@ -298,29 +263,29 @@ p.exchange-rate {
 }
 
 #exchange-rate {
-    color: #036574;
+    color: #006000;
     font-weight: bold;
 }
 
 .info {
-    display: flex;
+  display: flex;
 }
 
-/*@media only screen and (min-device-width: 320px) and (max-device-width: 480px) {
-    .info {
-        flex-direction: column;
-    }
-}*/
+@media only screen and (min-device-width : 320px) and (max-device-width : 480px) {
+  .info {
+    flex-direction: column;
+  }
+}
 
 .info > fieldset {
-    flex: 1;
-    padding: 1em;
-    border: 0;
+  flex: 1;
+  padding: 1em;
+  border: 0;
 }
 
 .info > div,
 .info > div fieldset {
-    height: 100%;
+  height: 100%;
 }
 
 div.info table {
@@ -331,429 +296,305 @@ div.info table td {
     vertical-align: top;
 }
 
-/*@media only screen and (min-device-width: 320px) and (max-device-width: 480px) {
-    div.info table td {
-        display: block;
-    }
+@media only screen and (min-device-width : 320px) and (max-device-width : 480px) {
+  div.info table td {
+    display: block;
+  }
 }
 
 #lp-info-container {
     display: none;
 }
 
-@media only screen and (min-device-width: 320px) and (max-device-width: 480px) {
-    .top-menu-bar > #poolsdropdown {
-        display: none;
-    }
-
-    #lp-info-container {
-        margin-top: 1em;
-    }
-
-    .border-menu {
-        display: block;
-        left: 50%;
-        transform: translateX(-50%);
-    }
-
-    .top-menu-bar a {
-        display: none;
-    }
-
-    #hamburger:checked ~ a, #hamburger:checked ~ div, #hamburger:checked ~ div > a {
-        display: block;
-    }
-
-    !*fix selector priority*!
-    .top-menu-bar.top-menu-bar #poolsdropdown:hover .dropdown {
-        display: none;
-    }
-
-    #poolsdropdown > a:after {
-        content: " Pools";
-    }
-}*/
-
-/*@media only screen and (max-device-width: 640px) {
-    span.poolstext, .pools {
-        width: auto;
-    }
-
-    span.pooltext {
-        display: inline-block;
-    }
+@media only screen and (min-device-width : 320px) and (max-device-width : 480px) {
+  .top-menu-bar > #poolsdropdown {
+    display: none;
+  }
+  #lp-info-container {
+    margin-top: 1em;
+  }
+  .border-menu {
+    display: block;
+    left: 50%;
+    transform: translateX(-50%);
+  }
+  .top-menu-bar a {
+    display: none;
+  }
+  #hamburger:checked ~ a, #hamburger:checked ~ div, #hamburger:checked ~ div > a {
+    display: block;
+  }
+  /*fix selector priority*/
+  .top-menu-bar.top-menu-bar #poolsdropdown:hover .dropdown {
+    display: none;
+  }
+  #poolsdropdown > a:after {
+    content: " Pools";
+  }
+}
+
+@media only screen and (max-device-width : 640px) {
+  span.poolstext, .pools {
+    width: auto;
+  }
+  
+  span.pooltext {
+    display: inline-block;
+  }
 
-    span.pools {
-        display: block;
-    }
-}*/
+  span.pools {
+    display: block;
+  }
+}
 
 .icon-container {
-    height: 15px;
-    position: relative;
+  height: 15px;
+  position: relative;
 }
 
-.border-menu {
-    position: relative;
-    border-radius: 2px;
-    height: 10px;
+.border-menu { 
+    position: relative;         
+    border-radius: 2px; 
+    height: 10px; 
     width: 10px;
     cursor: pointer;
     height: 20px;
     padding: 6px;
     text-align: left;
 }
-
-.border-menu:before {
+.border-menu:before { 
     border-bottom: 10px double black;
-    border-top: 3px solid black;
-    content: "";
-    height: 4px;
-    position: absolute;
-    width: 30px;
-}
-
-/*@media only screen and (min-device-width: 480px) {
-    .border-menu {
-        display: none;
-    }
-}*/
-
-legend {
-    padding: 0 !important;
+    border-top: 3px solid black;    
+    content:"";
+    height: 4px; 
+    position: absolute; 
+    width:30px;
 }
 
-#lp-info-currency {
-    text-align: left !important;
-}
-
-legend {
-    text-align: left;
+@media only screen and (min-device-width : 480px) {
+  .border-menu {
+    display: none;
+  }
 }
 
 .poolsdialog legend {
-    text-align: left;
+  text-align: center;
 }
 
 .poolsdialog a {
-    display: inline-block;
+  display: inline-block;
 }
-
 .poolsdialog > div {
-    cursor: pointer;
+  cursor: pointer;
 }
-
-.poolsdialog > div > span {
-    padding-right: 2em;
+.poolsdialog > div > span{
+  padding-right: 2em;
 }
-
 /* fixing selector priority */
 .window .poolsdialog a:hover {
-    text-decoration: none;
+  text-decoration: none;
 }
 
 #poolsdropdown {
-    display: inline-block;
-    cursor: pointer;
-    position: relative;
+  display: inline-block;
+  cursor: pointer;
+  position: relative;
 }
 
 .top-menu-bar #poolsdropdown:hover {
-    background-color: blue;
-    color: white;
+  background-color: blue;
+  color: white;
 }
 
 .top-menu-bar #poolsdropdown:hover .dropdown {
-    display: block;
+  display: block;
 }
 
 .dropdown {
-    display: none;
-    position: absolute;
-    min-width: 200px;
-    left: 50%;
-    transform: translateX(-50%);
-    background: #c0c0c0;
-    z-index: 1;
-    box-shadow: 0 0 0 3px silver, 1em 1em 3px 0 rgba(0, 0, 0, .5);
+  display: none;
+  position: absolute;
+  min-width: 200px;
+  left: 50%;
+  transform: translateX(-50%);
+  background: #c0c0c0;
+  z-index: 1;
+  box-shadow: 0 0 0 3px silver, 1em 1em 3px 0 rgba(0,0,0,.5);
 }
 
 #poolsdropdown > a {
-    margin: 0;
+  margin: 0;
 }
 
 .dropdown > a {
-    display: block;
-    margin: 0;
-    padding-left: 0.5em;
-    text-align: left;
+  display: block;
+  margin: 0;
+  padding-left: 0.5em;
+  text-align: left;
 }
 
 .simple-error, .success, .info-message {
-    margin: 1em auto 0 auto;
-    padding: 1em;
-    display: none;
-    color: white;
-    width: 80%;
+  margin: 1em auto 0 auto;
+  padding: 1em;
+  display: none;
+  color: white;
+  width: 80%;
 }
 
 .simple-error {
-    background: red;
-    border-radius: 8px;
+  background: red;
 }
 
 .info-message {
-    background: gray;
-    border-radius: 8px;
+  background: gray;
 }
 
 .success {
-    background: green;
-    border-radius: 8px;
+  background: green;
 }
 
 .pooltext {
-    display: inline-block;
-    width: 12%;
+  display: inline-block;
+  width: 12%;
 }
 
 .pools {
-    display: inline-block;
-    width: 60%;
+  display: inline-block;
+  width: 60%;
 }
 
 .loading {
-    display: inline-block;
-    overflow: hidden;
-    height: 1.3em;
-    margin-top: -0.3em;
-    line-height: 1.5em;
-    vertical-align: text-bottom;
+  display: inline-block;
+  overflow: hidden;
+  height: 1.3em;
+  margin-top: -0.3em;
+  line-height: 1.5em;
+  vertical-align: text-bottom;
 }
 
 .loading::after {
-    display: inline-table;
-    white-space: pre;
-    text-align: left;
+  display: inline-table;
+  white-space: pre;
+  text-align: left;
 }
 
 .loading.line::after {
-    content: "/\A–\A\\\A|";
-    text-align: center;
-    animation: spin 1s steps(4) infinite;
+  content: "/\A–\A\\\A|";
+  text-align: center;
+  animation: spin 1s steps(4) infinite;
 }
 
-@keyframes spin {
-    to {
-        transform: translateY(-6.0em);
-    }
+@keyframes spin  { 
+  to { 
+    transform: translateY( -6.0em); 
+  }
 }
 
 .text-center {
-    text-align: center;
+  text-align: center;
 }
 
 .lp-info-container {
-    display: none;
+  display: none;
 }
 
 .loading.matrix::after {
-    content: "░ ░░░░\A░░ ░░░\A░░░ ░░\A░░░░ ░";
-    animation: spin .5s steps(4) infinite alternate;
+  content:"░ ░░░░\A░░ ░░░\A░░░ ░░\A░░░░ ░";
+  animation: spin .5s steps(4) infinite alternate;
 }
 
 .loading.matrix {
-    font-size: 3em;
-    background: transparent;
-    border: none;
-    margin: 0.5em;
+  font-size: 3em;
+  background: transparent;
+  border: none;
+  margin: 0.5em;
 }
 
 .maxbalance {
-    padding: 0.3em;
-    text-align: right;
-    font-size: 0.95em;
-}
-
-fieldset {
-    border: none !important;
+  padding: 0.3em;
+  text-align: right;
+  font-size: 0.95em;
 }
 
 fieldset:nth-child(1) .maxbalance {
-    cursor: pointer;
+  cursor: pointer;
 }
 
 fieldset:nth-child(1) .maxbalance:hover {
-    text-decoration: underline;
+  text-decoration: underline;
 }
 
 fieldset:nth-child(2) .maxbalance {
-    visibility: hidden;
+  visibility: hidden;
 }
 
 .icon {
-    width: 1.5em;
-    height: 1.5em;
+  width: 1.5em;
+  height: 1.5em;
 }
 
 .ethicon {
-    filter: invert(0.4) sepia(1) saturate(0.8) hue-rotate(223deg);
+  filter: invert(0.4) sepia(1) saturate(0.8) hue-rotate(223deg);
 }
 
 .poolsdialog.donations > div {
-    margin-bottom: 3px;
+  margin-bottom: 3px;
 }
 
 .poolsdialog.donations > div > a {
-    display: flex;
-    align-items: center;
-    white-space: pre-wrap;
+  display: flex;
+  align-items: center;
+  white-space: pre-wrap;
 }
 
 .donations a {
-    word-break: normal;
+  word-break: normal;
 }
 
 .blue.window.info, .donations .text {
-    word-break: break-word;
+  word-break: break-word;
 }
 
 .donations .icon {
-    margin-left: 1em;
-    margin-right: 1em;
+  margin-left: 1em;
+  margin-right: 1em;
 }
 
 #custom_slippage_input {
-    width: 2.5em;
+  width: 2.5em;
 }
 
 #max_slippage {
-    margin-bottom: 10px;
-    text-align: left;
+  margin-bottom: 10px;
 }
 
-.exchange-rate {
-    margin-top: 3px;
-    margin-bottom: 10px;
+#max_slippage span {
+  padding-left: 4px;
 }
 
-.clearfix:after {
-    content: "";
-    display: table;
-    clear: both;
+#custom_slippage_input {
+  margin-left: -0.5em;
 }
 
-/*@media only screen and (min-device-width: 320px) and (max-device-width: 480px) {
-    p.exchange-rate {
-        text-align: left;
-    }
-
-    #max_slippage {
-        display: flex;
-        flex-wrap: wrap;
-    }
-
-    #max_slippage > * {
-        width: 100%;
-    }
-}*/
-
-
-[type="radio"]:checked,
-[type="radio"]:not(:checked) {
-    position: absolute;
-    left: -9999px;
-}
-[type="radio"]:checked + label,
-[type="radio"]:not(:checked) + label
-{
-    position: relative;
-    padding-left: 28px;
-    cursor: pointer;
-    line-height: 20px;
-    display: inline-block;
-    color: black;
+.exchange-rate {
+  margin-top: 3px;
+  margin-bottom: 10px;
 }
-[type="radio"]:checked + label:before,
-[type="radio"]:not(:checked) + label:before {
-    content: '';
-    position: absolute;
-    left: 0;
-    top: 0;
-    width: 18px;
-    height: 18px;
-    border: 1px solid #ddd;
-    border-radius: 100%;
-    background: #fff;
-}
-[type="radio"]:checked + label:after,
-[type="radio"]:not(:checked) + label:after {
-    content: '';
-    width: 12px;
-    height: 12px;
-    background: #036574;
-    position: absolute;
-    top: 4px;
-    left: 4px;
-    border-radius: 100%;
-}
-[type="radio"]:not(:checked) + label:after {
-    opacity: 0;
-    -webkit-transform: scale(0);
-    transform: scale(0);
-}
-[type="radio"]:checked + label:after {
-    opacity: 1;
-    -webkit-transform: scale(1);
-    transform: scale(1);
-}
-
-
-[type="checkbox"]:checked,
-[type="checkbox"]:not(:checked) {
-    position: absolute;
-    left: -9999px;
-}
-[type="checkbox"]:checked + label,
-[type="checkbox"]:not(:checked) + label
-{
-    position: relative;
-    padding-left: 28px;
-    cursor: pointer;
-    line-height: 20px;
-    display: inline-block;
-    color: black;
+
+.clearfix:after {
+  content: "";
+  display: table;
+  clear: both;
 }
-[type="checkbox"]:checked + label:before,
-[type="checkbox"]:not(:checked) + label:before {
-    content: '';
-    position: absolute;
-    left: 0;
-    top: 0;
-    width: 18px;
-    height: 18px;
-    border: 1px solid #ddd;
-    border-radius: 0;
-    background: #fff;
-}
-[type="checkbox"]:checked + label:after,
-[type="checkbox"]:not(:checked) + label:after {
-    content: '';
-    width: 12px;
-    height: 12px;
-    background: #036574;
-    position: absolute;
-    top: 4px;
-    left: 4px;
-    border-radius: 0%;
-}
-[type="checkbox"]:not(:checked) + label:after {
-    opacity: 0;
-    -webkit-transform: scale(0);
-    transform: scale(0);
-}
-[type="checkbox"]:checked + label:after {
-    opacity: 1;
-    -webkit-transform: scale(1);
-    transform: scale(1);
+
+@media only screen and (min-device-width : 320px) and (max-device-width : 480px) {
+  p.exchange-rate {
+    text-align: left;
+  }
+  #max_slippage {
+    display: flex;
+    flex-wrap: wrap;
+  }
+  #max_slippage > * {
+    width: 100%;
+  }
 }
\ No newline at end of file
diff --git a/withdraw.html b/withdraw.html
index 4c9e692..afa3872 100644
--- a/withdraw.html
+++ b/withdraw.html
@@ -21,13 +21,6 @@
     <link rel="icon" type="image/png" sizes="16x16" href="assets/favicon-16x16.png?rev=f30faa950b2b49eeffcaad8d774f6b10">
     <link rel="mask-icon" href="assets/safari-pinned-tab.svg" color="#5bbad5">
     <link rel="stylesheet" href="tvision.css?rev=3eda33f4fa54fe61a0ff034c823d8320" />
-      <script src="https://ajax.googleapis.com/ajax/libs/webfont/1.6.26/webfont.js" type="text/javascript"></script>
-      <script type="text/javascript">
-          WebFont.load({google: {families:
-                      ["Open Sans:300,300italic,400,400italic,600,600italic,700,700italic,800,800italic", "Nunito Sans:regular,700,900", "Nunito:regular,700,800,900"]
-              }});
-      </script>
-
       <script src="https://browser.sentry-cdn.com/5.13.0/bundle.min.js" integrity="sha384-ePH2Cp6F+/PJbfhDWeQuXujAbpil3zowccx6grtsxOals4qYqJzCjeIa7W2UqunJ" crossorigin="anonymous">
       </script>
       <script src='https://browser.sentry-cdn.com/5.13.0/captureconsole.min.js'></script>
@@ -43,20 +36,20 @@
   </head>
   <body>
     <div class="top-menu-bar">
-    <!--  <label for="hamburger" class='border-menu'></label>
-      <input type="checkbox" id="hamburger"/>-->
+      <label for="hamburger" class='border-menu'></label>
+      <input type="checkbox" id="hamburger"/>
 
       <a href="/">Buy and sell</a>
       <a href="/deposit.html">Deposit</a>
       <a href="/withdraw.html">Withdraw</a>
     </div>
     <div id="screen">
-        <div class="logo">
-            <h1><img src="assets/logo.png" alt="Harmony One" height="50"></h1>
+        <div class="blue window">
+            <h1><img src="assets/logo.svg" alt="🌀 Curve" height="50"></h1>
         </div>
         <div class="error window half-width info" id="error-window"></div>
 
-        <div class="window add-liquidity">
+        <div class="window white add-liquidity">
             <fieldset class="percentage">
                 <legend>Share of liquidity (%)</legend>
                 <ul>
@@ -96,10 +89,10 @@
         <fieldset id="lp-info-currency">
                 <legend>Currency reserves</legend>
                 <ul id='balances-info'>
-                    <li><b>USDT:</b> <span></span></li>
-                    <li><b>DAI:</b> <span></span></li>
-                    <li><b>BUSD:</b> <span></span></li>
-                    <li><b>USDT + DAI + BUSD:</b> <span></span></li>
+                    <li><b>Token A:</b> <span></span></li>
+                    <li><b>Token B:</b> <span></span></li>
+                    <li><b>Token C:</b> <span></span></li>
+                    <li><b>A + B + C:</b> <span></span></li>
                 </ul>
                 <p><b>Fee:</b> <span id='fee-info'></span>%</p>
                 <p><b>Admin fee:</b> <span id='admin-fee-info'></span>%</p>
@@ -107,10 +100,10 @@
               <fieldset id="lp-info-container">
                 <legend>My share:</legend>
                 <ul id='lp-info'>
-                    <li><b>USDT:</b> <span></span></li>
-                    <li><b>DAI:</b> <span></span></li>
-                    <li><b>BUSD:</b> <span></span></li>
-                    <li><b>USDT + DAI + BUSD:</b> <span></span></li>
+                    <li><b>Token A:</b> <span></span></li>
+                    <li><b>Token B:</b> <span></span></li>
+                    <li><b>Token C:</b> <span></span></li>
+                    <li><b>A + B + C:</b> <span></span></li>
                 </ul>
           </fieldset>
     </div>
diff --git a/withdraw.js b/withdraw.js
index f95d326..0c8e46a 100644
--- a/withdraw.js
+++ b/withdraw.js
@@ -64,7 +64,7 @@ function handle_change_amounts() {
                 $('[id^=currency_]').css('background-color', 'red');
             }
             else {
-                $('[id^=currency_]').css('background-color', '#036574');
+                $('[id^=currency_]').css('background-color', 'blue');
             }
             await calc_slippage(false);
 
@@ -85,7 +85,7 @@ function handle_change_share() {
     var val = share.val();
 
     console.log("handle change share")
-    share.css('background-color', '#036574');
+    share.css('background-color', 'blue');
     share.css('color', 'aqua');
     if (val == '---') {
         share.val('0.0');
@@ -125,8 +125,10 @@ async function handle_remove_liquidity() {
     // var default_account = (await web3provider.eth.getAccounts())[0];
     if (share_val == '---') {
         let token_amount = await SWAP.methods.calc_token_amount(amounts, false).call(CALL_OPTION);
-        console.log("token amount", token_amount.toString())
+
+        token_amount = token_amount.mul(BN(101)).div(BN(1000))
         console.log("wallet balance", token_balance.toString())
+        console.log("token amount", token_amount.toString())
         await SWAP.methods.remove_liquidity_imbalance(amounts, token_amount).send(CALL_OPTION);
     }
     else {
@@ -149,8 +151,9 @@ async function handle_remove_liquidity() {
 
 async function init_ui() {
     for (let i = 0; i < CONFIG.numCoins; i++) {
-        $('#currency_' + i).focus(handle_change_amounts(i));
-        $('#currency_' + i).on('input', debounced(100, handle_change_amounts(i)));
+        $('#currency_' + i).prop('disabled', true)
+        // $('#currency_' + i).focus(handle_change_amounts(i));
+        // $('#currency_' + i).on('input', debounced(100, handle_change_amounts(i)));
     }
     $('#liquidity-share').focus(handle_change_share);
     $('#liquidity-share').on('input', handle_change_share);
@@ -163,6 +166,7 @@ async function init_ui() {
 }
 
 window.addEventListener('load', async () => {
+
     await initWallet();
     await init_contracts()
     await update_rate_and_fees();
